MCNPInput.cpp

```diff
@@ -54,6 +54,7 @@ static int makeint( const std::string& token ){
   char* end;
   int ret = strtol(str, &end, 10);
   if( end != str+token.length() ){
+    record << "Warning: string [" << token << "] did not convert to int as expected." << std::endl;
     std::cerr << "Warning: string [" << token << "] did not convert to int as expected." << std::endl;
```

Comment:
:neckbeard: The ghost of hackers past moans:

These duplicated printouts make me sad.  Consider defining
```c
// Note: this macro will evaluate its argument expression twice!
#define NOISY_WARNING(OUTPUT_EXPR) do{ \
   std::cerr << "Warning: " << OUTPUT_EXPR << std::endl; \
   record    << "Warning: " << OUTPUT_EXPR << std::endl; \
} while (0)
 ```

Then these bug-prone output copypastas can become simply
```c
NOISY_WARNING( "string [" << token << "] did not convert to int as expected" );
```